packages/@aws-cdk/aws-iam/lib/lazy-role.ts

```diff
@@ -76,12 +76,12 @@ export class LazyRole extends cdk.Construct implements IRole {
   /**
    * Returns the ARN of this role.
    */
-  public get roleArn(): string {
-    return this.instantiate().roleArn;
+  public get attrArn(): string {
```

Comment:
No... we don't want to rename the L2s, just the L1s. The L2s should stay `roleArn`.